backend/src/authentication/authentication.controller.ts

```diff
@@ -14,14 +27,18 @@ export class AuthenticationController {
   public static readonly LOGIN_API_ROUTE = '/login';
   public static readonly LOGOUT_API_ROUTE = '/logout';
   public static readonly REFRESH_API_ROUTE = '/refresh';
+  public static readonly CHANGE_PASSWORD_API_ROUTE = '/change-password';
 
-  constructor(private readonly authenticationService: AuthenticationService) {}
+  constructor(
+    private readonly authenticationService: AuthenticationService,
+    private readonly userService: UserService
+  ) {}
 
-  @Public()
+  @Roles(Role.ADMIN)
```

Comment:
i created this `@Roles` decorator to restrict which users has access to this endpoint. for example, in this register users endpoint, only admins should be allowed to do this

the point of decorators themselves is to only set metadata about the method. there is no actual logic in the `@Roles` decorator itself. the logic is implemented in `RolesGuard`